net/src/main/java/io/bitcoinsv/jcl/net/protocol/handlers/message/MessageHandlerImpl.java

```diff
@@ -242,6 +248,24 @@ private void onStreamError(PeerAddress peerAddress, StreamErrorEvent event) {
         super.eventBus.publish(new DisconnectPeerRequest(peerAddress));
     }
 
+    private void onCorruptedData(PeerAddress peerAddress, StreamCorruptedDataEvent event) {
+        logger.trace(peerAddress, "Data corruption/error detected in Stream, requesting disconnection... ");
+        super.eventBus.publish(event); //re-publish the event to event bus (to allow further consumption & processing)
+        super.eventBus.publish(new DisconnectPeerRequest(peerAddress)); //also request to disconnect the peer - we probably do not want to keep p2p connection open with it since it sent corrupted data
```

Comment:
Shouldn't disconnect here as per discussion on daily.

Reply:
It was added because a StreamErrorEvent also causes a DisconnectPeerRequest to be published (this logic is currently present in main branch). We should then decide if the disconnect event is published in one, both or none of these cases.

Reply:
The line with the comment "//re-publish the event to event bus" re-publishes the event to the EventBus again. Have you checked if there is an infinite loop here? If you consume the event and publish it again, you should consume it again, publish it again, and so on....

Reply:
Testing the solution in our project, I did not encounter any loops (probably because these events are not registered inside the `registerForEvents` method). However, I did call `P2P`'s `getEventBus` method and subscribed to these events there. Should we improve this by implementing a `P2PEventStreamer` instance in `P2P` (something similar to `EVENTS`) for such errors and use that instead?

Reply:
I have refactored the solution a bit to follow the above mentioned idea. The re-publishing was removed from these methods. Please re-check.